src/lib/services/DependencyChecker.ts

```diff
@@ -0,0 +1,15 @@
+/**
+ * An interface for Dependency Checker engines.
+ * @author Josh Feingold
+ */
+import {DependencyResult} from '../../types';
+
+export interface DependencyChecker {
+	getName(): string;
+
+	run(target?: string): Promise<DependencyResult[]>;
```

Comment:
Is the `?` intentional?

Reply:
I figured it was better to leave parameters as open-ended as possible for now, and make it more specific as we flesh out the implementation and related commands.